### PR TITLE
JAVA-1948 Close session properly when LBP fails to initialize

### PR DESCRIPTION
**Pull request by GregBestland**

### CODE REVIEW

File: integration-tests/src/test/java/com/datastax/oss/driver/api/core/ConnectIT.java [hunk 1]
```diff
@@ -98,6 +100,34 @@ public void should_wait_for_contact_points_if_reconnection_enabled() throws Exce
...
+          .addContactPoints(simulacronRule.getContactPoints())
+          .withConfigLoader(loader)
+          .build();
+      fail("Should have thrown a DriverException for no DC with explicit contact point");
+    } catch (DriverException e) {
+    }
+    int size = 1;
+    long maxTime = System.currentTimeMillis() + 1000;
+    // One second should be plenty of time for connections to close server side
+    while (size > 0 && System.currentTimeMillis() < maxTime) {
```
Comment (adutra) [inline integration-tests/src/test/java/com/datastax/oss/driver/api/core/ConnectIT.java]: This is a very tight loop, you should at least sleep a little bit at each iteration. Or better yet, replace with:

``` 
    checkThat(() -> simulacronRule.cluster().getConnections().getConnections().isEmpty())
        .before(1, SECONDS)
        .becomesTrue();
```

File: integration-tests/src/test/java/com/datastax/oss/driver/api/core/ConnectIT.java [hunk 2]
```diff
@@ -98,6 +100,34 @@ public void should_wait_for_contact_points_if_reconnection_enabled() throws Exce
...
+      ClusterConnectionReport report = simulacronRule.cluster().getConnections();
+      size = report.getConnections().size();
+    }
+    assertThat(size).isEqualTo(0);
+  }
+
+  private static String provideNullString() {
```
Comment (adutra) [inline integration-tests/src/test/java/com/datastax/oss/driver/api/core/ConnectIT.java]: Unused.

File: integration-tests/src/test/java/com/datastax/oss/driver/api/core/ConnectIT.java [hunk 3]
```diff
@@ -98,6 +100,34 @@ public void should_wait_for_contact_points_if_reconnection_enabled() throws Exce
...
+      DriverConfigLoader loader =
+          SessionUtils.configLoaderBuilder()
+              .without(DefaultDriverOption.LOAD_BALANCING_LOCAL_DATACENTER)
+              .build();
+      CqlSession.builder()
+          .addContactPoints(simulacronRule.getContactPoints())
+          .withConfigLoader(loader)
+          .build();
+      fail("Should have thrown a DriverException for no DC with explicit contact point");
+    } catch (DriverException e) {
```
Comment (adutra) [inline integration-tests/src/test/java/com/datastax/oss/driver/api/core/ConnectIT.java]: Can you rename the exception to `ignored` to make IntelliJ happy? :)

File: integration-tests/src/test/java/com/datastax/oss/driver/api/core/ConnectIT.java [hunk 4]
```diff
@@ -98,6 +100,34 @@ public void should_wait_for_contact_points_if_reconnection_enabled() throws Exce
     session.close();
   }
 
+  @Test
+  public void should_cleanup_on_lbp_init_failure() throws Exception {
```
Comment (adutra) [inline integration-tests/src/test/java/com/datastax/oss/driver/api/core/ConnectIT.java]: Please add a little note in the javadocs of this method to link this test to JAVA-1948 for future reference.

**Review by adutra — CHANGES_REQUESTED** [review]

**Review by olim7t — CHANGES_REQUESTED** [review]
👍 nice way to fix this.

+1 to Alexandre's comments about the tests.

Please also add a changelog entry, and follow existing conventions for the message: affirmative form, no "should", colon after the ticket number, and no period at the end. E.g. "JAVA-1948: Close session properly when LBP fails to initialize"

**Review by olim7t — APPROVED** [review]

**Pull request merged**